PocketKit/Sources/PocketKit/Settings/AccountViewModel.swift

```diff
@@ -18,6 +26,23 @@ class AccountViewModel: ObservableObject {
         appSession.currentSession = nil
     }
 
+    func toggleAppBadge() {
+        UNUserNotificationCenter.current().requestAuthorization(options: .badge) {
+            (granted, error) in
+            guard error == nil && granted == true else {
+                self.userDefaults.set(false, forKey: "SettingsRowToggle")
+                DispatchQueue.main.async { [weak self] in
+                    self?.appBadgeToggle = false
+                }
+                return
+            }
+
+            let currentValue = self.userDefaults.bool(forKey: AccountViewModel.ToggleAppBadgeKey)
+            self.userDefaults.setValue(!currentValue, forKey: AccountViewModel.ToggleAppBadgeKey)
+            self.notificationCenter.post(name: .listUpdated, object: nil)
```

Comment:
Just a thought: `listUpdated` doesn't seem to fit within the context of exactly what's happening here, but I get why it's used. We _could_ add another notification for this to be more explicit, but… I don't think we have to right now.

Reply:
I had a comment here regarding this too, just curious what are some thoughts about adding a user defaults observer? 
https://github.com/Pocket/pocket-ios/pull/316#discussion_r1015591813